subsys/debug/coredump/CMakeLists.txt

```diff
@@ -23,6 +23,11 @@ zephyr_library_sources_ifdef(
   coredump_backend_flash_partition.c
   )
 
+zephyr_library_sources_ifdef(
+  CONFIG_DEBUG_COREDUMP_MW_BACKEND_INTEL_ADSP
```

Comment:
Zephyr folks, would CONFIG_DEBUG_COREDUMP_BACKEND_MW_INTEL_ADSP be better? That would seem to align better with existing backends.

Reply:
Or simply `CONFIG_DEBUG_COREDUMP_BACKEND_INTEL_ADSP`? Since this is going to be specific to Intel Audio DSP, how it's being done does not need to be encoded in the kconfig name, unless there are plans to use other transports for coredump info (now or in the near future).

Reply:
That is up to you. I've changed config and file names regarding with Kai's suggestion.

Reply:
I think `CONFIG_DEBUG_COREDUMP_BACKEND_INTEL_ADSP_MEM_WINDOW` might be better. This is more descriptive and easier to grep for `COREDUMP_BACKEND`.

Reply:
As you wish.